plonky3/src/prover/mod.rs

```diff
@@ -86,24 +86,68 @@ mod tests {
 
     /// Prove and verify execution
     fn run_test_goldilocks(pil: &str) {
+        run_test_goldilocks_publics(pil, None)
+    }
+
+    fn run_test_goldilocks_publics(pil: &str, malicious_publics: Option<Vec<GoldilocksField>>) {
         let mut pipeline = Pipeline::<GoldilocksField>::default().from_pil_string(pil.to_string());
 
         let pil = pipeline.compute_optimized_pil().unwrap();
         let witness_callback = pipeline.witgen_callback().unwrap();
         let witness = pipeline.compute_witness().unwrap();
 
-        let proof = Plonky3Prover::new(&pil).prove(&witness, witness_callback);
+        let prover = Plonky3Prover::new(&pil);
+        let proof = prover.prove(&witness, witness_callback);
 
         assert!(proof.is_ok());
+
+        if let Some(publics) = malicious_publics {
+            prover.verify(&proof.unwrap(), &[publics]).expect("")
+        }
     }
 
     #[test]
-    #[should_panic = "not implemented"]
     fn publics() {
         let content = "namespace Global(8); pol witness x; x * (x - 1) = 0; public out = x(7);";
         run_test_goldilocks(content);
     }
 
+    #[test]
+    fn public_inputs() {
+        let content = r#"
+        namespace Add(8);
+            col witness x;
+            col witness y;
+            col witness z;
+            y - 1 = 0;
+            x = 0;
+            x + y = z;
+
+            public outz = z(7);
+        "#;
+        let publics = Some(vec![GoldilocksField::from(1)]);
+
+        run_test_goldilocks_publics(content, publics);
+    }
```

Comment:
seems like these two test the same thing, let's keep only one. Also in this (honest) case, passing malicious public inputs is not needed.